playbooks/redhat.yml

```
@@ -19,6 +19,9 @@
         name: "{{ lookup('ansible.builtin.env', 'EDPM_SYSTEMROLES', default='redhat.rhel_system_roles') + '.rhc' }}"
         apply:
           become: true
-      when: ansible_facts['distribution'] | lower == 'redhat'
+      when:
+        - ansible_facts['distribution'] | lower == 'redhat'
+        - rhc_auth is defined
+        - rhc_auth | length > 0
       tags:
```

technically you can write this as:
```suggestion
        - rhc_auth | length > 0
      tags:
```

as the `| length` filter will already check if the item is defined.